bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -1773,6 +1813,131 @@ private CaseDbSchemaVersionNumber updateFromSchema8dot1toSchema8dot2(CaseDbSchem
 			releaseSingleUserCaseWriteLock();
 		}
 	}
+	
+	/**
+	 * Updates a schema version 8.2 database to a schema version 8.3 database.
+	 *
+	 * @param schemaVersion The current schema version of the database.
+	 * @param connection    A connection to the case database.
+	 *
+	 * @return The new database schema version.
+	 *
+	 * @throws SQLException     If there is an error completing a database
+	 *                          operation.
+	 * @throws TskCoreException If there is an error completing a database
+	 *                          operation via another SleuthkitCase method.
+	 */
+	private CaseDbSchemaVersionNumber updateFromSchema8dot2toSchema8dot3(CaseDbSchemaVersionNumber schemaVersion, CaseDbConnection connection) throws SQLException, TskCoreException {
+		if (schemaVersion.getMajor() != 8) {
+			return schemaVersion;
+		}
+
+		if (schemaVersion.getMinor() != 2) {
+			return schemaVersion;
+		}
+		
+		acquireSingleUserCaseWriteLock();
+		
+		ResultSet resultSet = null;
+		
+		try (Statement statement = connection.createStatement();) {
+			
+			// Add the uniqueness constraint to the tsk_event and tsk_event_description tables.
+			
+			// Unfortunately, SQLite doesn't support adding a constraint
+			// to an existing table so we have to rename the old...
+			
+			String primaryKeyType;
+			switch (getDatabaseType()) {
+				case POSTGRESQL:
+					primaryKeyType = "BIGSERIAL";
+					break;
+				case SQLITE:
+					primaryKeyType = "INTEGER";
+					break;
+				default:
+					throw new TskCoreException("Unsupported data base type: " + getDatabaseType().toString());
+			}
+			
+			
+			try{
```

Comment:
Schema 8.2 already creates a table tsk_event_types, and adds these 7 rows.  Does that need to be dropped first? Or does this need an "IF NOT EXISTS" clause?  

I can't tell how this tsk_event_types table differs from one in 8.2 (i.e. why re-create it all ?)

Reply:
I didn't know about the IF NOT EXISTS, I will added that and it might fix the bug you found.

Schema 8.2 did not include the tsk_event_types table.

The tsk_event_types table is only created if you upgraded from 8.1 to 8.2.  The code to add the tsk_event_types table was not added to the cpp code for the 8.2.

Therefore when you upgrade from 8.2 to 8.3 the table may or may not exist.